rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp

```diff
@@ -61,217 +72,564 @@ class LiaisonNegotiator : public rmf_traffic::schedule::Negotiator
+      id,
+      start_wp->index(),
+      finish_wp->index(),
+      cleaning_trajectory,
+      motion_sink,
+      ambient_sink,
+      tool_sink,
+      planner,
+      start_time,
+      drain_battery);
+
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Generated Clean request");
+  }
+
+  else if (task_type.type == rmf_task_msgs::msg::TaskType::TYPE_DELIVERY)
+  {
+    const auto& delivery = task_profile.delivery;
+    if (delivery.pickup_place_name.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [delivery.pickup_place_name] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    if (delivery.pickup_dispenser.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [delivery.pickup_dispenser] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    if (delivery.dropoff_place_name.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [delivery.dropoff_place_name] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    if (delivery.dropoff_place_name.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [delivery.dropoff_place_name] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    if (delivery.dropoff_ingestor.empty())
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Required param [delivery.dropoff_ingestor] missing in TaskProfile."
+        "Rejecting BidNotice with task_id:[%s]" , id.c_str());
+
+      return;
+    }
+
+    const auto pickup_wp = graph.find_waypoint(delivery.pickup_place_name);
+    if (!pickup_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), delivery.pickup_place_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    const auto dropoff_wp = graph.find_waypoint(delivery.dropoff_place_name);
+    if (!dropoff_wp)
+    {
+      RCLCPP_INFO(
+        node->get_logger(),
+        "Fleet [%s] does not have a named waypoint [%s] configured in its "
+        "nav graph. Rejecting BidNotice with task_id:[%s]",
+        name.c_str(), delivery.dropoff_place_name.c_str(), id.c_str());
+
+        return;
+    }
+
+    new_request = rmf_task::requests::Delivery::make(
+      id,
+      pickup_wp->index(),
+      delivery.pickup_dispenser,
+      dropoff_wp->index(),
+      delivery.dropoff_ingestor,
+      delivery.items,
+      motion_sink,
+      ambient_sink,
+      planner,
+      start_time,
+      drain_battery);
+
+    RCLCPP_INFO(
+      node->get_logger(),
+      "Generated Delivery request");
```

Comment:
Maybe add the `task_id` to this message for better transparency.

Reply:
Done https://github.com/osrf/rmf_core/pull/217/commits/f73f6896aedf9ba33247fe43c9034138912a848a